rclpy/rclpy/callback_groups.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2017 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from threading import Lock
+import weakref
+
+
+class CallbackGroup:
```

Comment:
Every class should subclass from `object` (if not from any other type).

Reply:
Even in python 3? [This page](https://docs.python.org/3.5/tutorial/classes.html#multiple-inheritance) says all classes inherit from object in python 3.5.

Reply:
I think you're right @sloretz we had the same interrogation before when developing rclpy and concluded that inheriting from object was not necessary, that's why non of the classes here inherit from object.

Reply:
We should follow that style in the existing code then.

Reply:
https://github.com/ros2/rclpy/pull/106 for rcllpy. I'll open PRs where necessary